### ISSUE TITLE
source code caching?

### ISSUE THREAD

**Issue by beckermr** [conda-forge/conda-forge.github.io#839]
I am hitting cases where source code from non-github/pypi sources is being either updated but with the same filename or is gone completely. Have we considered caching source code tarballs or enabling CF to do this if desired?

**Comment by jakirkham** [comment]
We have in some cases reuploaded these artifacts on the GitHub release of the feedstock.

**Comment by beckermr** [comment]
Interesting. I was thinking of something more automated where if the conda-forge.yaml has the right setting and the PR is a version bump, it downloads the artifact, uploads to someplace (could be the github release), and then makes a change to the recipe and makes a new PR.

**Comment by jakirkham** [comment]
There was a separate discussion about using split packages to handle this. So one artifact users would produce would be the source.

**Comment by bgruening** [comment]
Bioconda is using https://depot.galaxyproject.org/software/ (https://github.com/galaxyproject/cargo-port). Every tarball is mirrored. Are you considering to do this for every package as we do in Bioconda?

**Comment by bgruening** [comment]
University hosted infrastructure in PSU.

**Comment by sodre** [comment]
@isuruf, I think it would be awesome if we could store the tar-balls directly in anaconda.org under a different channel, e.g. `conda-forge-source` so we don't explode the repodata.json.

At the end of the day, we are looking for some Object Store, and I think we have a great one with anaconda.org that has an API we are all familiar.

**Comment by isuruf** [comment]
@sodre, you can only upload conda and pypi packages to anaconda.org.

**Comment by sodre** [comment]
We could wrap the tar.gz or source files in conda rather easily, wouldn't that satisfy the requirement?

I raised a similar issue to conda-build team about packaging models for Data Scientists. This is not that different. 

c.f.: https://github.com/conda/conda-build/pull/3529

**Comment by isuruf** [comment]
True. Do you have some script to do that? We can hook it up to conda-forge-ci-setup.

**Comment by isuruf** [comment]
That wouldn't work, IMO. It'll be harder to replace the file in the recipe

**Comment by sodre** [comment]
Let me run a couple of tests...

**Comment by jakirkham** [comment]
Where did we land on where the sources should be stored?
